gdal/gcore/gdalmultidim.cpp

```diff
@@ -3125,7 +3123,7 @@ static std::shared_ptr<GDALMDArray> CreateSlicedArray(
                 if( nDimSize < static_cast<GUInt64>(-endIdx) )
                     endIdx = 0;
                 else
```

Comment:
I think this could be more cleanly resolved by doing something like:

```
GUInt64 nEndIdx;
GIntBig endIdx = CPLAtoGIntBig(pszEnd);
if (endIdx < 0)
{
    nEndIdx = -endIdx;
    ...
}
else
    nEndIdx = endIdx;
```
This will eliminate a bunch of casts as well.

Reply:
Your proposed code is not equivalent to the intended logic, but I've simplified this a bit as
```
if( endIdx < 0 )
            {
                const auto positiveEndIdx = static_cast<GUInt64>(-endIdx);
                if( nDimSize < positiveEndIdx )
                    endIdx = 0;
                else
                    endIdx = nDimSize - positiveEndIdx;
            }
```